pkg/tfgen/docs.go

```diff
@@ -906,7 +905,9 @@ func (g *generator) convertHCL(hcl string) (string, string, error) {
 	convertHCL := func(languageName string) (err error) {
 		defer func() {
 			v := recover()
-			contract.Ignore(v)
+			if v != nil {
+				err = fmt.Errorf("panic converting HCL to %v: %v", languageName, v)
```

Comment:
It looks like we're losing some Go examples due to this. It appears to be picking up failures from gofmt, e.g.

```
invalid Go source code:

package main

import (
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
	"github.com/pulumi/pulumi-aws/sdk/v2/go/aws/cloudwatch"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		elasticsearch-log-publishing-policyPolicyDocument, err := iam.LookupPolicyDocument(ctx, &iam.LookupPolicyDocumentArgs{
			Statements: iam.getPolicyDocumentStatementArray{
				&iam.LookupPolicyDocumentStatement{
					Actions: []string{
						"logs:CreateLogStream",
						"logs:PutLogEvents",
						"logs:PutLogEventsBatch",
					},
					Principals: iam.getPolicyDocumentStatementPrincipalArray{
						&iam.LookupPolicyDocumentStatementPrincipal{
							Identifiers: []string{
								"es.amazonaws.com",
							},
							Type: "Service",
						},
					},
					Resources: []string{
						"arn:aws:logs:*",
					},
				},
			},
		}, nil);
		if err != nil {
			return err
		}
		elasticsearch-log-publishing-policyLogResourcePolicy, err := cloudwatch.NewLogResourcePolicy(ctx, "elasticsearch-log-publishing-policyLogResourcePolicy", &cloudwatch.LogResourcePolicyArgs{
			PolicyDocument: pulumi.String(elasticsearch-log-publishing-policyPolicyDocument.Json),
			PolicyName: pulumi.String("elasticsearch-log-publishing-policy"),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
```

In a sense, this is good: this is clearly bad codegen. However, it does make the improvements less clear. Manual inspection confirms that new examples are being picked up in the AWS schema.